spec/controllers/healthcheck_controller_spec.rb

```diff
@@ -0,0 +1,136 @@
+require 'rails_helper'
+
+RSpec.describe HealthcheckController, type: :controller do
+  render_views
+
+  let(:key) { 'super secret' }
+
+  around do |example|
+    original = Rails.configuration.metrics_auth_key
+    Rails.configuration.metrics_auth_key = key
+    example.call
+    Rails.configuration.metrics_auth_key = original
+  end
+
+  let(:parsed_body) {
+    JSON.parse(response.body)
+  }
+
+  let(:message_labs_address) { double }
+  let(:message_labs_port) { double }
+  let(:sendgrid_address) { double }
+  let(:sendgrid_port) { double }
+
+  before do
+    allow(SendgridHelper).to receive(:smtp_alive?).and_return(true)
+    allow(ZENDESK_CLIENT).to receive(:tickets).and_return(double(count: 0))
+  end
+
+  shared_examples 'a service is broken' do |service|
+    before do
+      get :index, key: key
+    end
+
+    it 'returns an HTTP Bad Gateway status' do
+      expect(response).to have_http_status(:bad_gateway)
+    end
+
+    it "reports #{service} as inaccessible" do
+      expect(parsed_body).to include('checks' => include(service => false))
+    end
+  end
```

Comment:
They were there before! I just pulled them out of the support file they were hidden in.